polaris-service/src/main/java/org/apache/polaris/service/admin/PolarisAdminService.java

```diff
@@ -525,6 +525,7 @@ private boolean catalogOverlapsWithExistingCatalog(CatalogEntity catalogEntity)
                                     if (newLocation == null || existingLocation == null) {
                                       return false;
                                     }
+                                    // TODO normalize the locations here
```

Comment:
There are a number of ways we could do this. We could handcraft some abstraction over URI, or we could use an existing library like Hadoop FileSystem or Apache VFS that automatically does this for us.

Reply:
For now, I added a lightweight implementation (`StorageLocation`) but I would like to treat this as an open door. I miss Hadoop's `FileSystem` :)

Reply:
Yeah we should really consider leveraging all the Path stuff from there actually. Normalizing/relativizing paths, parsing scheme+authority+path correctly, etc., are areas where we definitely don't want to perpetuate reinventing the wheel, especially if we are using lots of `startsWith`, `endsWith` etc., to attempt to collapse things like trailing slashes, match scheme, etc.

We'll need a broader refactor effort though since there's a lot of ad-hoc path handling scattered throughout the code right now.